agent-bootstrap/src/main/java/io/opentelemetry/auto/instrumentation/api/SpanWithScope.java

```diff
@@ -19,7 +19,59 @@
 import io.opentelemetry.context.Scope;
 import io.opentelemetry.trace.Span;
 
-// intentionally (for now) not implementing Closeable since not clear what it would close
+/**
+ * This is deprecated.
+ *
+ * <p>Originally, we used {@code SpanWithScope} to pass the {@link Span} and {@link Scope} between
+ * {@code @Advice.OnMethodEnter} and {@code @Advice.OnMethodExit}, e.g.
+ *
+ * <pre>
+ *   &#64;Advice.OnMethodEnter(...)
+ *     public static SpanWithScope onEnter(...) {
+ *     ...
+ *     Span span = ...
+ *     return new SpanWithScope(span, currentContextWith(span));
+ *   }
+ *
+ *   &#64;Advice.OnMethodExit(...)
+ *   public static void stopSpan(
+ *       ...
+ *       &#64;Advice.Enter final SpanWithScope spanWithScope) {
+ *     Span span = spanWithScope.getSpan();
+ *     ...
+ *     span.end();
+ *     spanWithScope.closeScope();
+ *   }
+ * </pre>
+ *
+ * But we are (slowly) migrating to a new style using {@code @Advice.Local} that doesn't require
```

Comment:
I think it deserves the explanation: why are we migrating? What was bad with old approach? Document your decisions, not actions :)